src/main/components/uml-element/resizable/resizable.tsx

```diff
@@ -60,39 +91,92 @@ export const resizable =
         return (
           <WrappedComponent {...props}>
             {props.children}
-            <Handle onPointerDown={this.onPointerDown} />
+            <HandleBottomRight
+              onPointerDown={(e) => {
+                this.onPointerDown(e, 'bottomRight');
+              }}
+            />
+            <HandleTopLeft
+              onPointerDown={(e) => {
+                this.onPointerDown(e, 'topLeft');
+              }}
+            />
+            <HandleTopRight
+              onPointerDown={(e) => {
+                this.onPointerDown(e, 'topRight');
+              }}
+            />
+            <HandleBottomLeft
+              onPointerDown={(e) => {
+                this.onPointerDown(e, 'bottomLeft');
+              }}
+            />
           </WrappedComponent>
         );
       }
 
-      private resize = (width: number, height: number) => {
+      private resize = (width: number, height: number, resizeFrom: string) => {
         width = Math.round(width / 20) * 20;
         height = Math.round(height / 20) * 20;
         if (options && options.preventX) width = 0;
         if (options && options.preventY) height = 0;
         if (width === 0 && height === 0) return;
 
         this.setState((state) => ({ offset: state.offset.add(width, height) }));
-        this.props.resize({ width, height });
+        this.props.resize({ width, height }, resizeFrom);
       };
 
-      private onPointerDown = (event: React.PointerEvent<SVGElement>) => {
+      private onPointerDown = (event: React.PointerEvent<SVGElement>, resizeFrom: string) => {
         if (event.nativeEvent.which && event.nativeEvent.which !== 1) {
           return;
         }
 
-        this.setState({ resizing: true, offset: new Point(event.clientX, event.clientY) });
+        switch (resizeFrom) {
```

Comment:
here i would create a `let offset` which we then assign in the switch statement.
Thus we can have one `this.setState({ resizing: true, offset })` in the end.

Reply:
exactly as you did in the onPointerMove function.

Reply:
@martindevtum 
Done.